arch/arm/core/aarch32/cortex_m/CMakeLists.txt

```diff
@@ -15,6 +15,16 @@ zephyr_library_sources(
 
 zephyr_library_sources_ifdef(CONFIG_DEBUG_COREDUMP coredump.c)
 
+if(CONFIG_ARCH_HAS_TIMING_FUNCTIONS AND
```

Comment:
I would remove this external if and _prioritize_ the ARCH-solution (DWT) if present, against SoC and Board.